drivers/kubevirt/kubevirt.go

```diff
@@ -507,3 +515,330 @@ func AddPVCsToVirtualMachine(vm kubevirtv1.VirtualMachine, pvcs []*corev1.Persis
 
 	return nil
 }
+
+// StartAndWaitForVMIMigration starts the VM migration and waits for the VM to be in running state in the new node
+func StartAndWaitForVMIMigration(virtualMachineCtx *scheduler.Context, ctx context1.Context, schedDriver scheduler.Driver) error {
+
+	vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{virtualMachineCtx})
+	if err != nil {
+		return err
+	}
+
+	// Get the namespace and name of the VM
+	if len(vms) == 0 {
+		return fmt.Errorf("no VMs found for VM [%s] in namespace [%s]", virtualMachineCtx.App.Key, virtualMachineCtx.App.NameSpace)
+	}
+	vmiNamespace := vms[0].Namespace
+	vmiName := vms[0].Name
+	if len(vms) > 1 {
```

Comment:
Please move this check one line above

Reply:
Done added a single check, which passes only if len(vms)=1 else fail.